creator-node/src/RehydrateIpfsQueue.js

```diff
@@ -0,0 +1,95 @@
+const Bull = require('bull')
+const config = require('./config')
+const { rehydrateIpfsFromFsIfNecessary, rehydrateIpfsDirFromFsIfNecessary } = require('./utils')
+const { logger: genericLogger } = require('./logging')
+
+const PROCESS_NAMES = Object.freeze({
+  rehydrate_dir: 'rehydrate_dir',
+  rehydrate_file: 'rehydrate_file'
+})
+
+class RehydrateIpfsQueue {
+  constructor () {
+    this.queue = new Bull(
+      'rehydrateIpfs',
+      {
+        redis: {
+          host: config.get('redisHost'),
+          port: config.get('redisPort')
+        }
+      }
+    )
+
+    this.queue.process(PROCESS_NAMES.rehydrate_file, config.get('rehydrateMaxConcurrency'), async (job, done) => {
+      const { multihash, storagePath, filename, logContext } = job.data
+
+      this.logStatus(logContext, `RehydrateIpfsQueue - Processing a rehydrateIpfsFromFsIfNecessary task for ${multihash}`)
+      // TODO: all the expected rehydrate errors are caught/logged. so catch might not actually catch error
+      // can instead of logging errors in rehydrate code, throw errors and handle here
+      try {
+        await rehydrateIpfsFromFsIfNecessary(multihash, storagePath, logContext, filename)
+        done()
+      } catch (e) {
+        this.logStatus(logContext, `RehydrateIpfsQueue - Problem with processing a rehydrateIpfsFromFsIfNecessary task for ${multihash}: ${e}`)
```

Comment:
if this is an error log we should handle that appropriately in logStatus. our loggly has filters based on INFO vs ERROR so want to keep that accurate